packages/cli/src/databases/repositories/test-definition.repository.ee.ts

```diff
@@ -0,0 +1,60 @@
+import type { FindManyOptions, FindOptionsWhere } from '@n8n/typeorm';
+import { DataSource, In, Repository } from '@n8n/typeorm';
+import { Service } from 'typedi';
+
+import { TestDefinition } from '@/databases/entities/test-definition.ee';
+import type { ListQuery } from '@/requests';
+
+@Service()
+export class TestDefinitionRepository extends Repository<TestDefinition> {
+	constructor(dataSource: DataSource) {
+		super(TestDefinition, dataSource.manager);
+	}
+
+	async getMany(sharedWorkflowIds: string[], options?: ListQuery.Options) {
+		if (sharedWorkflowIds.length === 0) return { tests: [], count: 0 };
+
+		const where: FindOptionsWhere<TestDefinition> = {
+			...options?.filter,
+			workflow: {
+				id: In(sharedWorkflowIds),
+			},
+		};
+
+		const findManyOptions: FindManyOptions<TestDefinition> = {
+			where,
+			relations: ['annotationTag'],
+			order: { createdAt: 'DESC' },
+		};
+
+		if (options?.take) {
+			findManyOptions.skip = options.skip;
+			findManyOptions.take = options.take;
+		}
+
+		const [testDefinitions, count] = await this.findAndCount(findManyOptions);
+
+		return { testDefinitions, count };
+	}
+
+	async getOne(id: number, sharedWorkflowIds: string[]) {
+		return await this.findOne({
+			where: {
+				id,
+				workflow: {
+					id: In(sharedWorkflowIds),
+				},
+			},
+			relations: ['annotationTag'],
+		});
+	}
+
+	async deleteById(id: number, sharedWorkflowIds: string[]) {
+		return await this.delete({
+			id,
+			workflow: {
```

Comment:
I'm not too familiar with the idiom here, but what does this signify? Is this a where clause? why is it relevant to select the workflows when deleting by identifier?

Reply:
It's a condition to make sure user can delete only tests he have access to (tests, where workflow under test is among the workflows accessible by user)